bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/template/TemplateManager.java

```diff
@@ -0,0 +1,118 @@
+ */
+public class TemplateManager {
+
+    private ServiceTracker templateProviderTracker;
+
+    /**
+     * @param bc
+     */
+    public TemplateManager(BundleContext bc) {
+        templateProviderTracker = new ServiceTracker(bc, TemplateProvider.class.getName(), null);
+        templateProviderTracker.open();
+    }
+
+    public Template getTemplate(String templateUID) {
+        return getTemplate(templateUID, null);
+    }
+
+    public Template getTemplate(String templateUID, Locale locale) {
+        Template template = null;
+        Object[] providers = templateProviderTracker.getServices();
+        for (int i = 0; providers != null && i < providers.length; i++) {
+            template = ((TemplateProvider) providers[i]).getTemplate(templateUID, locale);
+            if (template != null) {
+                return createCopy(template);
+            }
+        }
+        return null;
+    }
+
+    public Collection<Template> getTemplatesByTag(String tag) {
+        return getTemplatesByTag(tag, null);
+    }
+
+    public Collection<Template> getTemplatesByTag(String tag, Locale locale) {
+        Collection<Template> result = new ArrayList<Template>(20);
+        Collection<Template> templates = null;
+        Object[] providers = templateProviderTracker.getServices();
+        for (int i = 0; providers != null && i < providers.length; i++) {
+            templates = ((TemplateProvider) providers[i]).getTemplates(locale);
+            if (templates != null) {
+                for (Iterator<Template> it = templates.iterator(); it.hasNext();) {
+                    Template t = it.next();
+                    if (tag != null) {
+                        Collection<String> tags = t.getTags();
+                        if (tags != null && tags.contains(tag)) {
+                            result.add(createCopy(t));
+                        }
+                    } else {
+                        result.add(createCopy(t));
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    public Collection<Template> getTemplates() {
+        return getTemplates(null);
+    }
+
+    public Collection<Template> getTemplates(Locale locale) {
+        return getTemplatesByTag(null, locale);
+    }
+
+    public void dispose() {
+        templateProviderTracker.close();
+        templateProviderTracker = null;
+    }
+
+    private Template createCopy(Template template) {
+        if (template == null) {
+            return null;
+        }
+        if (template instanceof RuleTemplate) {
+            RuleTemplate rt = (RuleTemplate) template;
+            RuleTemplate t = new RuleTemplate(template.getUID(),//
+                    rt.getModules(Trigger.class), //
+                    rt.getModules(Condition.class), //
+                    rt.getModules(Action.class), //
+                    rt.getConfigurationDescription(), rt.getVisibility());
+            t.setTags(rt.getTags());
```

Comment:
I would say no. We had discussed that templates are only provided through providers. If they are read from bundles as JSON, you would have the problem of where to persist such changes - you cannot write them back into the bundles, so you would have to use a database and then you would have to sync these two versions somehow.